Frontend/library/src/Inputs/MouseController.ts

```diff
@@ -253,56 +120,122 @@ export class MouseController {
+        this.videoElementParent.removeEventListener('mousedown', this.onMouseDownListener);
+        this.videoElementParent.removeEventListener('mouseup', this.onMouseUpListener);
+        this.videoElementParent.removeEventListener('contextmenu', this.onContextMenuListener);
+        this.videoElementParent.removeEventListener('wheel', this.onMouseWheelListener);
+        this.videoElementParent.removeEventListener('dblclick', this.onMouseDblClickListener);
+
+        super.unregisterMouseEvents();
+    }
+
+    private onMouseDown(event: MouseEvent) {
+        if (!this.videoPlayer.isVideoReady()) {
+            return;
+        }
+        const coord = this.coordinateConverter.normalizeAndQuantizeUnsigned(event.offsetX, event.offsetY);
+        this.streamMessageController.toStreamerHandlers.get('MouseDown')([event.button, coord.x, coord.y]);
+        event.preventDefault();
+    }
+
+    private onMouseUp(event: MouseEvent) {
+        if (!this.videoPlayer.isVideoReady()) {
             return;
         }
-        const toStreamerHandlers = this.toStreamerMessagesProvider.toStreamerHandlers;
-        toStreamerHandlers.get('MouseEnter')();
+        const coord = this.coordinateConverter.normalizeAndQuantizeUnsigned(event.offsetX, event.offsetY);
+        this.streamMessageController.toStreamerHandlers.get('MouseUp')([event.button, coord.x, coord.y]);
+        event.preventDefault();
     }
 
-    /**
-     * Handles mouse Leave
-     */
-    sendMouseLeave() {
-        if (!this.videoElementProvider.isVideoReady()) {
+    private onContextMenu(event: MouseEvent) {
+        if (!this.videoPlayer.isVideoReady()) {
             return;
         }
-        const toStreamerHandlers = this.toStreamerMessagesProvider.toStreamerHandlers;
-        toStreamerHandlers.get('MouseLeave')();
+        event.preventDefault();
     }
 
-    /**
-     * Handles when a mouse button is pressed down
-     * @param button - Mouse Button Pressed
-     * @param X  - Mouse X Coordinate
-     * @param Y  - Mouse Y Coordinate
-     */
-    sendMouseDown(button: number, X: number, Y: number) {
-        if (!this.videoElementProvider.isVideoReady()) {
+    private onMouseMove(event: MouseEvent) {
+        if (!this.videoPlayer.isVideoReady()) {
             return;
         }
-        Logger.Info(`mouse button ${button} down at (${X}, ${Y})`);
-        const toStreamerHandlers = this.toStreamerMessagesProvider.toStreamerHandlers;
-        toStreamerHandlers.get('MouseDown')([button, X, Y]);
+        const coord = this.coordinateConverter.normalizeAndQuantizeUnsigned(event.offsetX, event.offsetY);
+        const delta = this.coordinateConverter.normalizeAndQuantizeSigned(event.movementX, event.movementY);
+        this.streamMessageController.toStreamerHandlers.get('MouseMove')([
+            coord.x,
+            coord.y,
+            delta.x,
+            delta.y
+        ]);
+        event.preventDefault();
```

Comment:
I question preventDefault() on mousemove (I understand this is directly ported).

My reason for question it is because preventDefault() will cause the next event to be delayed by a pretty decent ms (like 50ms I have read).